src/software/ai/hl/stp/tactic/receiver/receiver_fsm.cpp

```diff
@@ -108,26 +104,28 @@ std::optional<Shot> ReceiverFSM::findFeasibleShot(const World& world,
 bool ReceiverFSM::onetouchPossible(const Update& event)
 {
     return !event.control_params.disable_one_touch_shot &&
+           !receiver_tactic_config.disable_one_touch_kick() &&
            (findFeasibleShot(*event.common.world_ptr, event.common.robot) !=
             std::nullopt);
 }
 
 void ReceiverFSM::updateOnetouch(const Update& event)
 {
     auto best_shot = findFeasibleShot(*event.common.world_ptr, event.common.robot);
-    auto one_touch = getOneTouchShotPositionAndOrientation(
-        event.common.robot, event.common.world_ptr->ball(),
-        best_shot->getPointToShootAt());
 
-    if (best_shot && event.control_params.pass)
+    if (best_shot.has_value() && event.control_params.pass)
     {
+        auto one_touch = getOneTouchShotPositionAndOrientation(
+            event.common.robot, event.common.world_ptr->ball(),
+            best_shot->getPointToShootAt());
+
         event.common.set_primitive(std::make_unique<MovePrimitive>(
             event.common.robot, one_touch.getPointToShootAt(), one_touch.getOpenAngle(),
             TbotsProto::MaxAllowedSpeedMode::PHYSICAL_LIMIT,
             TbotsProto::ObstacleAvoidanceMode::AGGRESSIVE, TbotsProto::DribblerMode::OFF,
             TbotsProto::BallCollisionType::ALLOW,
             AutoChipOrKick{AutoChipOrKickMode::AUTOKICK,
-                           BALL_MAX_SPEED_METERS_PER_SECOND}));
+                           BALL_MAX_SPEED_METERS_PER_SECOND - 0.5}));
```

Comment:
maybe MovePrimitive should add auto-padding instead of us remembering to add it in a bunch of places

Reply:
Okay that's a good idea. I've kept this the same, but Im now clamping the value in `MovePrimitive` as well.